### DISCUSSION TITLE
Modélisation applicative

### DISCUSSION THREAD

**Discussion by VinceBlood** [dbarzin/mercator#138] [Q&A]
Bonjour,

J'aimerai avoir un petit éclaircissements sur la modélisation des applications.

Nous avons plusieurs applications type client lourd.
Ces dernières communiquent avec :
- une base de données en direct
- des services (Serveur Appli Win32) de manière indirecte en passant par un broker de messages (ActiveMQ)

Comment faut-il modéliser ce type d'architecture logicielle ?

Dans les faits, l'application et le serveur applicatif sont connectés sur le services broker de message.

Par ailleurs vous confirmez qu'un service API-Rest est considéré comme un service et non une application ?
Et qu'un module serait par exemple JAVA soutenant le service API-Rest ?

Enfin l'application lourde s'installe sur des postes de travail, or il semblerait qu'il ne soit pas possible d'associer d'application sur un poste de travail.

Concernant les bases de données, doit-on les déclarer comme des "Applications", "Services" en plus de "Base de données" ?

**Comment by dbarzin** [comment]
Bonsoir Vincent,

La modélisation a adopter doit être celle qui répond à tes attentes en utilisant ce que Mercator permet.
Plusieurs possibilités de modélisation existent, à toi de choisir celle qui convient le mieux.

Le lien entre applications et postes de travail va être ajouté.

Pour les bases de données, il faut au moins les déclarer comme base de données afin de pouvoir faire le lien avec les données.
Il peut être nécessaire de les déclarer également comme application. Cela dépend du niveau de modélisation que tu souhaites obtenir.

**Comment by dbarzin** [comment]
Bonsoir Vincent,
Le lien entre applications et postes de travail a être ajouté.
Il est également possible de déclarer la configuration des postes de travail.
Didier

**Comment by Arnaud-Askee** [comment]
Bonjour,
Dans le même esprit, je cherche à faire un lien entre la base de données (Vue application) et le stockage de données (Vue Infra physique).
Est-ce possible ?

**Comment by dbarzin** [comment]
L'ajout d'un lien entre la base de données (Vue application) et serveur logique (Vue logique) ainsi qu'un lien entre un serveur logique (Vue logique) et une infrastructure de stockage (Vue infra physique).
Un lien direct entre la base de données et l'infrastructure de stockage ne va pas permettre d'identifier le type de base de données et ses caractéristiques.

**Reply by Arnaud-Askee** [reply]
Les caractéristiques sont portées au bon endroit actuellement. Mon problème est dans le lieu de stockage de la base (Sur quel serveur physique). J'ai imaginé que cette information pouvait être portée par ce lien entre ces 2 composants de ces 2 vues.

Selon ma compréhension, on peut avoir : 
Application -> Serveur Logique -> Serveur Physique
J'imagine pouvoir déclarer de la même manière
Base de donnée -> Stockage de donnée -> Serveur Physique

(Le lien application / BDD étant déjà géré).

Comment peut-on gérer sans ce lien ?

**Comment by dbarzin** [comment]
Tu proposes d'ajouter des liens entre :
- la table "base de données" et "infrastructure de stockage"
- la table "infrastructure de stockage" et la table "serveur physique".

Le premier lien passerai au dessus de la vue de l'infrastructure logique en faisant un lien entre la vue applicative et la vue plysique, cela ne permetrai pas d'identifier l'OS, l'éditeur de la base de données.

Je me demande s'il ne faudrait pas fusioner les tables "infrastructure de stockage" et "serveur physique" ?
